### DISCUSSION TITLE
What is the mkdocs material version of sphinx toctree?

### DISCUSSION THREAD

**Discussion by ssbarnea** [squidfunk/mkdocs-material#5200] [Q&A] [answered]
I am looking for a way to include a toctree inside the page body like we have with [sphinx toctree](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html) but so far I was not able to spot that on mkdocs side. Any hints?

Update, I found it also posted on https://stackoverflow.com/questions/57280455/mkdocs-how-to-create-table-of-contents-from-mkdocs-yml-inside-index-md

**Comment by squidfunk** [comment, marked as answer]
Short summary of the answer: there's (currently) no plugin or extension to print the entire navigation, but it's possible for the table of contents on the current page by including the `[TOC]` marker in the Markdown file at the desired position.

**Reply by ssbarnea** [reply]
I was able to make `[TOC]` work as a decent replacement, but I had to add some custom configuration to it:
```
  - toc:
      toc_depth: 2-3
      permalink: true
```

The key was the `2-` part which prevents inclusion of current page title inside the generated toc, as that would generate an undesired duplication of the title. I hope that would help others looking for the same.
